### DISCUSSION TITLE
Is there any way to get online users?

### DISCUSSION THREAD

**Discussion by mopo3ula** [keycloak/keycloak#23161] [Q&A] [answered]
Hello! I faced the problem where I need to get all online users or at least their IDs. In the documentation I found the route user-sessions but it returns paginated list of all sessions and if user has multiple sessions I get it multiple times . I got like 500k sessions so I can't get all the sessions and paginate them by myself.

For me it is necessary to get a list of non-repeating users.

Is there any way to do this? Thank you in advance!

**Comment by justin-tay** [comment, marked as answer]
I think the closest REST API for your requirements is the `sessions` endpoint that is used by the admin console exposed by  `keycloak-rest-admin-ui-ext`. You can try to set a high `max` value. But this still will list multiple sessions of the same user.

`http://keycloak/admin/realms/{realm}/ui-ext/sessions?first=0&max=999999&type=REGULAR&search=`

I think if you really need a list of non-repeating users you will have to write a custom REST API https://www.keycloak.org/docs/latest/server_development/#_extensions_rest.

You can look at the existing implementation as a basis to write your custom endpoint to get the distinct IDs.

https://github.com/keycloak/keycloak/blob/f767eb109e4d419a08be2bb09c42e43b706ee7b2/rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java#L61-L96

**Reply by mopo3ula** [reply]
Thank you for your answer! I will take a look on how fast this request is completed and if it takes a long time then I will try to write a cusom REST API.
